### PR TITLE
Fx/case study core tools kits card highlights 

### PR DESCRIPTION
**Pull request by kareraolivier**
- [x] This task contain the highlight on cards when you select it in highlight level and tech.

### CODE REVIEW

File: src/components/HightlightCaseStudyDetails.tsx [hunk 1]
```diff
@@ -1,56 +1,62 @@
...
-      {entries.map((element, index) => (
-        <button
-          key={index}
-          className={`border border-gray-500 p-4 rounded-lg ${style}`}
-        >
-          {element}
-        </button>
-      ))}
-    </div>
+const HightlightsBotton = (
```
Comment (NNesta) [inline src/components/HightlightCaseStudyDetails.tsx:9]: correct this typo here

File: src/components/HightlightCaseStudyDetails.tsx [hunk 2]
```diff
@@ -1,56 +1,62 @@
...
+      </button>
+    ))}
   </div>
 );
 
-const HightlightCaseStudyDetails = ({ levels, technologies }: Props) => {
+const HightlightCaseStudyDetails = ({ setHightlightedElement }: Props) => {
+  const getCaseStudyDetails = (element: string) => {
+    setHightlightedElement(element);
+  };
```
Comment (NNesta) [inline src/components/HightlightCaseStudyDetails.tsx:30]: You can remove this  function and use setHigtligtedElement only

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/components/HightlightCaseStudyDetails.tsx (hunk 1 above)
Comment (kareraolivier) [reply src/components/HightlightCaseStudyDetails.tsx:9]: fixed

File: src/components/HightlightCaseStudyDetails.tsx (hunk 2 above)
Comment (kareraolivier) [reply src/components/HightlightCaseStudyDetails.tsx:30]: removed

**Review by NNesta — APPROVED** [review]

File: src/data/case-studies/index.ts [hunk 3]
```diff
@@ -1,27 +1,64 @@
...
+      {
+        title: "Case Studies",
+        entries: caseStudies,
+      },
+    ],
+  },
 ] as unknown as Group[];
-export default groups;
+
+const getEntryKeys = (highlightedElement: any) =>
```
Comment (musayann) [inline src/data/case-studies/index.ts:34]: The data files should only contain the data and shouldn't contain these helper classes. please move these out of this file

**Review by musayann — CHANGES_REQUESTED** [review]

File: src/data/case-studies/index.ts [hunk 4]
```diff
@@ -1,27 +1,64 @@
...
+    )
+    .map((entry) => entry.Key);
+
+const Highlights = [
+  {
+    title: "Highlight Case Study Details",
+    blocks: [
+      {
+        title: "Case Study Implementation Level",
+        style: "bg-black-900 text-white",
```
Comment (musayann) [inline src/data/case-studies/index.ts:53]: these shouldn't be in these data files

**Review by musayann — CHANGES_REQUESTED** [review]

File: src/contex/index.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useState } from "react";
+
+export type ContextType = {
+  entryKeys: string[];
+  setEntryKeys: any;
+};
+const AppContext = createContext<ContextType | undefined>(undefined);
+
+export function ContextProvider({ children }: { children: React.ReactNode }) {
+  const [entryKeys, setEntryKeys] = useState([]);
```
Comment (musayann) [inline src/contex/index.tsx:10]: the name entry keys is not clear enough

File: src/pages/case-studies-and-core-tools.tsx [hunk 6]
```diff
@@ -1,24 +1,40 @@
...
 import Group from "@/components/Group";
-import groups from "@/data/case-studies";
-import HightlightCaseStudyDetails from "@/components/HightlightCaseStudyDetails";
-import { levels } from "@/data/case-studies/levels";
-import { technologies } from "@/data/case-studies/technologies";
+import HighlightCaseStudyDetails from "@/components/HightlightCaseStudyDetails";
+
 const caseStudiesAndCoreTools = () => {
+  const [highlightedElement, setHighlightedElement] = useState<string>("");
+  const { setEntryKeys } = useAppContext();
```
Comment (musayann) [inline src/pages/case-studies-and-core-tools.tsx:11]: The logic used in highlighting an element in the co-design journey should be the same as that used for case studies highlight

**Review by musayann — CHANGES_REQUESTED** [review]

File: src/data/case-studies/index.ts (hunk 3 above)
Comment (kareraolivier) [reply src/data/case-studies/index.ts:34]: removed

File: src/data/case-studies/index.ts (hunk 4 above)
Comment (kareraolivier) [reply src/data/case-studies/index.ts:53]: removed

Comment (kareraolivier) [reply src/data/case-studies/index.ts:53]: removed

File: src/pages/case-studies-and-core-tools.tsx (hunk 6 above)
Comment (kareraolivier) [reply src/pages/case-studies-and-core-tools.tsx]: changed

File: src/contex/index.tsx (hunk 5 above)
Comment (kareraolivier) [reply src/contex/index.tsx]: changed

**Pull request closed without merging**